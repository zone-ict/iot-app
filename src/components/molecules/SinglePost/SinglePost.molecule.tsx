import React from 'react';
import { Post } from '../../../models/Post.model';

interface Props {
  data?: Post;
  isLoading: boolean;
}

export default function SinglePost({ data, isLoading }: Props) {
  if (isLoading || !data) {
    return <div>Loading...</div>;
  }

  return <div>{data.title}</div>;
}
