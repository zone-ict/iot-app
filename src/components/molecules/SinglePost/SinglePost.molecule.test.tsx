import React from 'react';
import renderer from 'react-test-renderer';
import { Post } from '../../../models/Post.model';
import SinglePost from './SinglePost.molecule';

const testPost: Post = {
  id: 1,
  title: 'test',
  body: 'test',
  userId: 1,
};

describe('Single post component', () => {
  it('should be able to render correctly', () => {
    const tree = renderer.create(<SinglePost isLoading={false} data={testPost} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to render loading view correctly', () => {
    const tree = renderer.create(<SinglePost isLoading data={testPost} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should be able to render loading view when there's no data", () => {
    const tree = renderer.create(<SinglePost isLoading={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
