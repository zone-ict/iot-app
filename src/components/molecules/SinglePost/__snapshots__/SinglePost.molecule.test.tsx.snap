// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single post component should be able to render correctly 1`] = `
<div>
  test
</div>
`;

exports[`Single post component should be able to render loading view correctly 1`] = `
<div>
  Loading...
</div>
`;

exports[`Single post component should be able to render loading view when there's no data 1`] = `
<div>
  Loading...
</div>
`;
